src/App.js

```diff
@@ -44,21 +46,32 @@ export default class App extends React.Component {
 		}
 	};
 	toggleTheme = () => {
-		let theme = !this.state.on;
+		const theme = !this.state.on;
 		this.setState({ on: theme });
 		this.setInStorage('theme', theme);
 	};
 	addToPhotos = (photo) => {
-		let savedPhotos = [...this.state.savedPhotos, photo];
-		this.setState({ savedPhotos: savedPhotos });
-		this.setInStorage('savedPhotos', savedPhotos);
+		if (
+			!this.state.savedPhotos.find((savedPhoto) => savedPhoto.id === photo.id)
+		) {
+			const savedPhotos = [...this.state.savedPhotos, photo];
+			this.setState({ savedPhotos: savedPhotos });
+			this.setInStorage('savedPhotos', savedPhotos);
```

Comment:
here, you want to stringify

Reply:
I had already stringified the setInStorage method on line 33